compiler/ast.nim

```diff
@@ -1430,6 +1430,12 @@ proc newSymNode*(sym: PSym, info: TLineInfo): PNode =
   result.typ = sym.typ
   result.info = info
 
+proc newSymNode*(sym: PSym, info: TLineInfo, typ: PType): PNode =
+  result = newNode(nkSym)
+  result.sym = sym
+  result.typ = typ
+  result.info = info
```

Comment:
So now you introduced the notion of a nkSym node where `n.typ != n.sym.typ`. Unworkable.

Reply:
so assign sym.typ to typ? here the typ is distinct base type or the param.typ, doesn't make much differience here.

Reply:
Maybe use `nkConv`?

Reply:
not sure, previous arguments are passed as `nkEmpty` node, but `nkEmpty` is used for `errorNode`, maybe make up fails cases show the problem.